internal/services/postgres/postgresql_flexible_server_resource.go

```diff
@@ -382,7 +382,6 @@ func resourcePostgresqlFlexibleServer() *pluginsdk.Resource {
 			}
 
 			newMb = newStorageMbRaw.(int)
-			newTier = newTierRaw.(string)
 
```

Comment:
I assume that expandArmServerStorage also needs to be update since d.GetOk("storage_tier") would always gets the value in the previous apply when it isn't set in the tf config. It's unexpected. Below is the sample for the fix.

```
// storage_tier is added `Computed: true` so that TF always returns the value in the previous apply when `storage_tier` isn't set in the tf config. So we need to use `IsExplicitlyNullInConfig` to check if it's set in the tf config
	if !pluginsdk.IsExplicitlyNullInConfig(d, "storage_tier") {
		if v, ok := d.GetOk("storage_tier"); ok {
			storage.Tier = pointer.To(servers.AzureManagedDiskPerformanceTiers(v.(string)))
		}
	}
```